solr/solr-ref-guide/src/common-query-parameters.adoc

```diff
@@ -278,9 +278,12 @@ If you do not define the `wt` parameter in your queries, JSON will be returned a
 
 Solr caches the results of all queries and filter queries by default. To disable result caching, set the `cache=false` parameter.
 
-You can also use the `cost` option to control the order in which non-cached filter queries are evaluated. This allows you to order less expensive non-cached filters before expensive non-cached filters.
+You can also use the `cost` option to provide a hint as to the order in which non-cached filter queries are evaluated.
+This allows you to order less expensive non-cached filters before expensive non-cached filters.
+At the Lucene layer, this maps to `TwoPhaseIterator.matchCost` if the query has a TPI.
 
-For very high cost filters, if `cache=false` and `cost>=100` and the query implements the `PostFilter` interface, a Collector will be requested from that query and used to filter documents after they have matched the main query and all other filter queries. There can be multiple post filters; they are also ordered by cost.
+*Post Filters*: For very high cost filters, if `cache=false` _and_ `cost>=100`, _and_ the query implements the `PostFilter` interface, a Collector will be requested from that query and used to filter documents after they have matched the main query and all other filter queries.
```

Comment:
This is a bigger issue than you probably intend to deal with here, and I recognize you're editing an existing sentence, but I want to mention it as something to consider - there are 5 total mentions of PostFilters in the ref guide and none of them define what a "post filter" is or when or how or what specifically implements a "PostFilter interface".

I mention it because I was looking at this section the other day and wondered how I would be able to know if my query implements a PostFilter - is it a thing? a method? a backend default? Here and following we're emphasizing the relationship between the `cache` and `cost` params and a PostFilter and this expects the reader to know what it is. But we've never defined it anywhere.

As I said it's a bigger problem than your small edit, but since I don't know what it is in order to define it anywhere, maybe you'd have time to improve this gap?

Reply:
I'd like to improve the situation further; thanks for sharing your insights.

Last night I observed that cache & cost are poorly placed on this page; they are *not* top level request parameters; they are "local parameters" which are not even defined on this page (another problem).  These two parameters would best be documented beneath "fq" (filter query) documentation and maybe referenced from another page that defines the caches if such exists.

RE "Post Filter" -- definitely an advanced subject.  I'm doing changes for 9.x that will further reduce the instances of a PostFilter to those few that really need to be one.  Outside of this paragraph, the only other instance of "PostFilter" I see is on the spatial page.  It is within the scope of this PR to remove that reference because it's no longer interesting that LLPSF does _not_ implement PostFilter.  What else am I missing?

Reply:
> These two parameters would best be documented beneath "fq" (filter query) documentation and maybe referenced from another page that defines the caches if such exists.

I agree. I think it would be in scope of this PR to move it? Or not, up to you.

> Outside of this paragraph, the only other instance of "PostFilter" I see is on the spatial page.

The docs on collapse mention that the CollapsingQParser is a post filter, as a way of introducing it (in italics even), but again does not define what it is or provide any context for why that matters. For most users that means absolutely nothing when they don't know what a post filter is to begin with. Of course, dealing with that is out of scope for this PR, but it is another place where the term is just dropped in without any ability for anyone to learn what it is.